CONTRIBUTING.md

```diff
@@ -1,107 +1,174 @@
+`auto-tooling-and-instrumentation.isolated`. In addition, the extension of
+all class files is changed from `class` to `classdata`. This ensures that
+general classloaders cannot find nor load these classes.
+- When `io.opentelemetry.auto.bootstrap.Agent` starts up, it creates an
+instance of `io.opentelemetry.auto.bootstrap.AgentClassLoader`, loads an
+`io.opentelemetry.auto.tooling.AgentInstaller` from that `AgentClassLoader`
+and then passes control on to the `AgentInstaller` (now in the
+`AgentClassLoader`). The `AgentInstaller` then installs all of the
+instrumentations with the help of ByteBuddy.
+
+The complicated process above ensures that the majority of
+auto-instrumentation agent's classes are totally isolated from application
+classes, and an instrumented class from arbitrary classloader in JVM can
+still access helper classes from bootstrap classloader.
 
 #### Agent jar structure
-If you now look inside `java-agent/build/libs/opentelemetry-auto-<version>.jar`, 
-you will see the following "clusters" of classes:
-* `auto-tooling-and-instrumentation.isolated/` - contains `agent-tooling` module and 
-`instrumentation` submodules, loaded and isolated inside `AgentClassLoader`. 
-Including OpenTelemetry SDK.
-* `io/opentelemetry/auto/bootstrap/` - contains `agent-bootstrap` module and available in
-bootstrap classloader.
-* `io/opentelemetry/auto/shaded/` - contains OpenTelemetry API and its dependencies. 
-Shaded during creation of `java-agent` jar file by Shadow Gradle plugin.
+
+If you now look inside
+`java-agent/build/libs/opentelemetry-auto-<version>.jar`, you will see the
+following "clusters" of classes:
+
+- `auto-tooling-and-instrumentation.isolated/` - contains `agent-tooling`
+module and `instrumentation` submodules, loaded and isolated inside
+`AgentClassLoader`. Including OpenTelemetry SDK.
+- `io/opentelemetry/auto/bootstrap/` - contains `agent-bootstrap` module and
+available in bootstrap classloader.
+- `io/opentelemetry/auto/shaded/` - contains OpenTelemetry API and its
+dependencies. Shaded during creation of `java-agent` jar file by Shadow
+Gradle plugin.
+
+### Building
+
+#### Snapshot builds
+
+For developers testing code changes before a release is complete, there are
+snapshot builds of the `master` branch. When a PR is merged to `master`, a
+circleci build is kicked off as a github action which shows up as a github
+check on the git commit on `master` branch, i.e. a green checkmark. Clicking
+on the green checkmark you can view the `build_test_deploy` workflow and the
+`build` job shows the artifacts hosted on circleci. The artifacts will be
+named like:
+
+```
+libs/exporter-support-<version>-SNAPSHOT.jar
+libs/opentelemetry-auto-<version>-SNAPSHOT.jar
+libs/opentelemetry-auto-exporters-jaeger-<version>-SNAPSHOT.jar
+libs/opentelemetry-auto-exporters-logging-<version>-SNAPSHOT.jar
+libs/opentelemetry-auto-exporters-otlp-<version>-SNAPSHOT.jar
+```
+
+#### Building from source
+
+Build using Java 8:
```

Comment:
You need java 11 to build current repo

Reply:
good catch 👍, i can fix this tomorrow if u don't beat me to it